application/comit_node/src/comit_client/mod.rs

```diff
@@ -31,10 +31,18 @@ pub trait ClientFactory<C: Client>: Send + Sync + RefUnwindSafe + Debug {
 
 #[derive(Clone, Debug, PartialEq, Eq)]
 pub enum SwapReject {
+    Declined {
+        reason: SwapDeclineReason,
+    },
     /// The counterparty rejected the request
     Rejected,
 }
 
+#[derive(Clone, Debug, PartialEq, Eq, Serialize, Deserialize)]
+pub enum SwapDeclineReason {
+    Unspecified,
```

Comment:
Shall we implement few reasons the user can pass in the `POST` request?
Also, is there some existing `internal error` cases that we could clean up?
When do we use `Rejected`? is that internal errors only?
I think under `Rejected` we could also have an enum. Something like:
```
pub enum SwapReject {
    Declined {
        pub enum Reason {
            Liquidity,
            Rate,
            Unspecified,
            AlphaTimeout,
            BetaTimeout,
        }
    },
    Rejected {
        pub enum Reason {
            UnsupportedLedger,
            InternalError,
        }
    }
}
```

Reply:
Maybe worth doing a quick meeting/brainstorm about it.

Reply:
@D4nte I think that idea has potential.

Reply:
#583 opened to track